specs/directory.md

```diff
@@ -0,0 +1,167 @@
+# Notation Directory Structure
+
+The `notation` CLI requires local file systems support for the following components across various platforms.
+
+- `notation` binary
+- Plugins
+- Configurations
+- Trust stores
+- Trust policies
+- Signature caches
+- Local key store
+
+This documentation specifies the recommended directory structure for those components.
+
+## Category
+
+The directories for various components are classified into the following catagories.
+
+| Alias     | Description                                                                         |
+| --------- | ----------------------------------------------------------------------------------- |
+| `BIN`     | Directory for executable binaries                                                   |
+| `LIBEXEC` | Directory for binaries not meant to be executed directly by users' shell or scripts |
+| `CONFIG`  | Directory for configurations                                                        |
+| `CACHE`   | Directory for user-specific cache                                                   |
+
+Although it is recommended to install `notation` with its plugins and default configurations at the system level, it is possible to install at the user level.

```

Comment:
Why we want user to install `notation` with its plugins and default configurations at the system level? Isnt user level more secure and isolated?

Reply:
`notation` may come with some default plugins or company-specific shared plugins.

For instance, `docker` has `docker-app` and `docker-buildx` bundled when `docker` CLI is installed.

Reply:
Of course, plugins can be installed at the user level.